### DISCUSSION TITLE
How can I prevent automatic advertising after disconnection?

### DISCUSSION THREAD

**Discussion by sivar2311** [h2zero/NimBLE-Arduino#372] [Q&A] [answered]
I send out an advertisement with service data for 5 seconds. (`pBleAdvertisement->start(5)`). 
Within this time, it is possible for other devices to establish a connection and read out the characteristics.

If no device establishes a connection, the advertisement stops (as expected) after about 5 seconds.

But if a device establishes a connection and then disconnects, the advertisement starts again - this time of unlimited duration. 
How can I prevent this?

Edit:
I also noticed that after a device was connected and I start advertising manufacturer data (see #371) - which is sent out for 1 second (`pBleAdvertisement->start(1)`), the duration is completely ignored. The advertisement runs forever.

**Comment by h2zero** [comment, marked as answer]
The function you're looking for is here:
https://h2zero.github.io/NimBLE-Arduino/class_nim_b_l_e_server.html#a6bfd923ecd0ea06d5564343ab7209122

>I also noticed that after a device was connected and I start advertising manufacturer data (see https://github.com/h2zero/NimBLE-Arduino/discussions/371) - which is sent out for 1 second (pBleAdvertisement->start(1)), the duration is completely ignored. The advertisement runs forever.

Strange, I'll check it out.

**Reply by sivar2311** [reply]
Thank you very much! 
That's what i was looking for!
`NimBLEServer::advertiseOnDisconnect(false)` fixed also the endless advertising!
